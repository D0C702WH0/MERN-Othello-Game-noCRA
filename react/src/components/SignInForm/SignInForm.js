import React, { Component } from 'react'
import { Form, Button, Card, Container } from 'react-bootstrap'
import { ToastContainer } from 'react-toastify'
import { Redirect } from 'react-router-dom'
import PropTypes from 'prop-types'
import '../../app.scss'

export default class SignInForm extends Component {
  state = {
    password: '',
    pseudo: '',
    redirectToUserHome: false
  }

  static getDerivedStateFromProps (nextProps, prevState) {
    const { auth } = nextProps
    const isUserLoggedIn = auth
    return auth === { redirectToUserHome: prevState } ? { redirectToUserHome: prevState } : { redirectToUserHome: isUserLoggedIn }
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  handleSubmit (e) {
    e.preventDefault()
    const { pseudo, password } = this.state
    const user = {
      pseudo,
      password
    }
    this.props.logIn(user)
  }

  hasInvalidFields = () => {
    const { password, pseudo } = this.state
    if (password.trim() !== '' && pseudo.trim() !== '') {
      return false
    }
    return true
  }

  render () {
    const { pseudo, password, redirectToUserHome } = this.state
    const { handleChange, handleSubmit } = this
    const isDisabled = this.hasInvalidFields() ? true : null

    if (redirectToUserHome === true) {
      return (<Redirect to="/"/>)
    }
    return (
      <div style={{ margin: '3rem' }}>
        <ToastContainer autoClose={false} />
        <Container>
          <h2>Connexion</h2>
          <Card className="forms">
            <Form onSubmit={handleSubmit.bind(this)}>
              <Form.Group controlId="formBasicPseudo">
                <Form.Label>Pseudo</Form.Label>
                <Form.Control onChange={handleChange.bind(this)} name="pseudo" value={pseudo} type="text" placeholder="Pseudo" />
              </Form.Group>
              <Form.Group controlId="formBasicPassword">
                <Form.Label>Mot de passe</Form.Label>
                <Form.Control onChange={handleChange.bind(this)} name="password" value={password} type="password" placeholder="Mot de passe" />
              </Form.Group>
              <Button variant="secondary" type="submit" disabled={isDisabled}>
              Valider
              </Button>
            </Form>
          </Card>
        </Container>
      </div>
    )
  }
}

SignInForm.propTypes = {
  logIn: PropTypes.func,
  auth: PropTypes.bool
}
