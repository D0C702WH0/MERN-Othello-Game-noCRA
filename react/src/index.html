<!DOCTYPE html>
<html lang="fr">
  <head>
    <title>zoOthello</title>
    <base href="/"/>
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="stylesheet" href="style.css"/>
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css"
    integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous" />
    <link href="https://fonts.googleapis.com/css?family=Patua+One&display=swap" rel="stylesheet"/>    
    <link href="https://fonts.googleapis.com/css?family=Roboto&display=swap" rel="stylesheet"/>
  </head>
  <body>
    <div id="root"></div>
  </body>
</html>