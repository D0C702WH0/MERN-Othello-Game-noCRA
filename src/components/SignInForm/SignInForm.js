import React, { Component } from 'react'
import { Form, Button, Card, Container } from 'react-bootstrap'
import { Redirect } from 'react-router-dom'
import PropTypes from 'prop-types'

export default class SignInForm extends Component {
  state = {
    password: '',
    pseudo: '',
    redirectToUserHome: false
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  handleSubmit (e) {
    e.preventDefault()
    const { pseudo, password } = this.state
    const user = {
      pseudo,
      password
    }
    this.props.logIn(user)
  }

  render () {
    const { pseudo, password, redirectToUserHome } = this.state
    const { handleChange, handleSubmit } = this
    if (redirectToUserHome) {
      return (<Redirect to="/home/user"/>)
    }
    return (
      <Container>
        <Card>
          <Form onSubmit={handleSubmit.bind(this)}>
            <Form.Group controlId="formBasicPseudo">
              <Form.Label>Pseudo</Form.Label>
              <Form.Control onChange={handleChange.bind(this)} name="pseudo" value={pseudo} type="text" placeholder="Pseudo" />
            </Form.Group>
            <Form.Group controlId="formBasicPassword">
              <Form.Label>Mot de passe</Form.Label>
              <Form.Control onChange={handleChange.bind(this)} name="password" value={password} type="password" placeholder="Password" />
            </Form.Group>
            <Button variant="primary" type="submit">
              Valider
            </Button>
          </Form>
        </Card>
      </Container>
    )
  }
}

SignInForm.propTypes = {
  logIn: PropTypes.func
}
