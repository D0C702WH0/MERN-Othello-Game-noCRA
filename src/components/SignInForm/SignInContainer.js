import { connect } from 'react-redux'
import { logIn } from '../../actions/authActions'
import SignInForm from './SignInForm'

const mapStateToProps = ({ auth }) => ({
  auth
})

const mapDispatchToProps = (dispatch) => ({
  logIn: (user) => {
    dispatch(logIn(user))
  }
})

const SignInContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(SignInForm)

export default SignInContainer
