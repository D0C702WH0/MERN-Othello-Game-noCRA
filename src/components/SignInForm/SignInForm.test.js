import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import SignInForm from './SignInForm'

describe('<SignInForm />', () => {
  let wrapper

  beforeEach(() => {
    wrapper = render(
      <SignInForm />
    )
  })

  it('renders', () => {
    const { asFragment } = wrapper
    expect(asFragment()).toMatchSnapshot()
  })

  it('it has a disabled button until both pseudo and password field have a value', () => {
    const pseudo = 'pseudo'
    const password = 'password'

    const { getByPlaceholderText, getByText } = wrapper

    const submitButton = getByText('Valider')
    expect(submitButton.disabled).toEqual(true)

    const pseudofieldNode = getByPlaceholderText('Pseudo')
    fireEvent.change(pseudofieldNode, { target: { value: pseudo } })
    expect(submitButton.disabled).toEqual(true)

    const passwordfieldNode = getByPlaceholderText('Mot de passe')
    fireEvent.change(passwordfieldNode, { target: { value: password } })
    expect(passwordfieldNode.value).toEqual(password)
    expect(pseudofieldNode.value).toEqual(pseudo)
    expect(submitButton.disabled).toEqual(false)
  })
})
