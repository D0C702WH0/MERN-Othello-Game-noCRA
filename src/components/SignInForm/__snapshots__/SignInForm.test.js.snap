// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInForm /> renders 1`] = `
<DocumentFragment>
  <div
    style="margin: 3rem;"
  >
    <div
      class="Toastify"
    />
    <div
      class="container"
    >
      <h2>
        Connexion
      </h2>
      <div
        class="forms card"
      >
        <form
          class=""
        >
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="formBasicPseudo"
            >
              Pseudo
            </label>
            <input
              class="form-control"
              id="formBasicPseudo"
              name="pseudo"
              placeholder="Pseudo"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="formBasicPassword"
            >
              Mot de passe
            </label>
            <input
              class="form-control"
              id="formBasicPassword"
              name="password"
              placeholder="Mot de passe"
              type="password"
              value=""
            />
          </div>
          <button
            class="btn btn-secondary"
            disabled=""
            type="submit"
          >
            Valider
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
